clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -678,6 +686,13 @@ invariant_checker_t::check_for_pc_discontinuity(per_shard_t *shard,
                 if (!valid_branch_flow) {
                     error_msg = "Direct branch does not go to the correct target";
                 }
+            } else if (cur_instr_decoded != nullptr &&
+                       shard->prev_instr_decoded_ != nullptr &&
+                       shard->prev_instr_.instr.addr != 0 /*first*/ &&
```

Comment:
We already check `shard->prev_instr_.instr.addr != 0`  in the if-condition at L647, so this condition is not needed again here.

Reply:
Done.